libraries/fbx/src/GLTFSerializer.cpp

```diff
@@ -979,668 +1074,504 @@ bool GLTFSerializer::buildGeometry(HFMModel& hfmModel, const hifi::VariantHash&
+                    } else {
+                        mesh.clusterIndices.push_back(0);
+                        mesh.clusterIndices.push_back(0);
+                        mesh.clusterIndices.push_back(0);
                     }
                 }
-
-                if (primitive.defined["material"]) {
-                    part.materialID = materialIDs[primitive.material];
+            } else if (primitiveAttributes.contains("JOINTS_0")) {
+                for (int i = 0; i < partVerticesCount; ++i) {
+                    for (int j = 0; j < 4; ++j) {
+                        mesh.clusterIndices.push_back(0);
+                    }
                 }
-                mesh.parts.push_back(part);
+            }
 
-                // populate the texture coordinates if they don't exist
-                if (mesh.texCoords.size() == 0 && !hfmModel.hasSkeletonJoints) {
-                    for (int i = 0; i < part.triangleIndices.size(); ++i) {
-                        mesh.texCoords.push_back(glm::vec2(0.0, 1.0));
+            if (!mesh.clusterIndices.empty()) {
+                int skinIndex = gltfMeshToSkin[gltfMeshIndex];
+                if (skinIndex != -1) {
+                    const auto& deformer = hfmModel.skinDeformers[(size_t)skinIndex];
+                    std::vector<uint16_t> oldToNew;
+                    oldToNew.resize(_file.nodes.size());
+                    for (uint16_t clusterIndex = 0; clusterIndex < deformer.clusters.size() - 1; ++clusterIndex) {
+                        const auto& cluster = deformer.clusters[clusterIndex];
+                        oldToNew[(size_t)cluster.jointIndex] = clusterIndex;
                     }
                 }
+            }
 
-                // Build morph targets (blend shapes)
-                if (!primitive.targets.isEmpty()) {
+            // populate the texture coordinates if they don't exist
+            if (mesh.texCoords.size() == 0 && !hfmModel.hasSkeletonJoints) {
+                for (int i = 0; i < part.triangleIndices.size(); ++i) {
+                    mesh.texCoords.push_back(glm::vec2(0.0, 1.0));
+                }
+            }
 
-                    // Build list of blendshapes from FST and model.
-                    typedef QPair<int, float> WeightedIndex;
-                    hifi::VariantHash blendshapeMappings = mapping.value("bs").toHash();
-                    QMultiHash<QString, WeightedIndex> blendshapeIndices;
+            // Build morph targets (blend shapes)
+            if (!primitive.targets.isEmpty()) {
 
-                    for (int i = 0;; ++i) {
-                        auto blendshapeName = QString(BLENDSHAPE_NAMES[i]);
-                        if (blendshapeName.isEmpty()) {
-                            break;
-                        }
-                        auto mappings = blendshapeMappings.values(blendshapeName);
-                        if (mappings.count() > 0) {
+                // Build list of blendshapes from FST and model.
```

Comment:
this blendshape stuff was another conflict but I tried to maintain the code in master from here: https://github.com/kasenvr/project-athena/pull/671